ydb/public/sdk/cpp/src/client/impl/ydb_internal/grpc_connections/grpc_connections.h

```diff
@@ -272,6 +272,150 @@ class TGRpcConnectionsImpl
+
+                dbState->StatCollector.IncGRpcInFlight();
+                dbState->StatCollector.IncGRpcInFlightByHost(endpoint.GetEndpoint());
+
+                NYdbGrpc::TAdvancedResponseCallback<TResponse> responseCbLow =
+                    [this, context, userResponseCb = std::move(userResponseCb), endpoint, dbState]
+                    (const grpc::ClientContext& ctx, TGrpcStatus&& grpcStatus, TResponse&& response) mutable -> void {
+                        dbState->StatCollector.DecGRpcInFlight();
+                        dbState->StatCollector.DecGRpcInFlightByHost(endpoint.GetEndpoint());
+
+                        if (NYdbGrpc::IsGRpcStatusGood(grpcStatus)) {
+                            std::multimap<std::string, std::string> metadata;
+
+                            for (const auto& [name, value] : ctx.GetServerInitialMetadata()) {
+                                metadata.emplace(
+                                    std::string(name.begin(), name.end()),
+                                    std::string(value.begin(), value.end()));
+                            }
+                            for (const auto& [name, value] : ctx.GetServerTrailingMetadata()) {
+                                metadata.emplace(
+                                    std::string(name.begin(), name.end()),
+                                    std::string(value.begin(), value.end()));
+                            }
+
+                            auto resp = new TResult<TResponse>(
+                                std::move(response),
+                                std::move(grpcStatus),
+                                std::move(userResponseCb),
+                                this,
+                                std::move(context),
+                                endpoint.GetEndpoint(),
+                                std::move(metadata));
+
+                            EnqueueResponse(resp);
+                        } else {
+                            dbState->StatCollector.IncReqFailDueTransportError();
+                            dbState->StatCollector.IncTransportErrorsByHost(endpoint.GetEndpoint());
+
+                            auto resp = new TGRpcErrorResponse<TResponse>(
+                                std::move(grpcStatus),
+                                std::move(userResponseCb),
+                                this,
+                                std::move(context),
+                                endpoint.GetEndpoint());
+
+                            dbState->EndpointPool.BanEndpoint(endpoint.GetEndpoint());
+
+                            EnqueueResponse(resp);
+                        }
+                    };
+
+                serviceConnection->DoAdvancedRequest(*request, std::move(responseCbLow), rpc, meta,
+                    context.get());
+            }, dbState, requestSettings.PreferredEndpoint, requestSettings.EndpointPolicy);
+    }
+
+
```

Comment:
I literally copy-paste the implementation of `Run` here except for a single change in the parameter type: accept `TRequest*` instead of `TRequest&&`. It is needed to avoid copying of the request inside `Run` inside `WithServiceConnection`; due to the nature it arena-allocated messages, `std::move` does copying when trying to move from an arena-allocated message into a non-arena-allocated message to preserve correct ownership (see "Generated Message Class" section [here](https://protobuf.dev/reference/cpp/arenas/#usage)).

Same for `RunDeferredOnArena` below.